packages/perspective-viewer/src/js/viewer/perspective_element.js

```diff
@@ -267,6 +267,14 @@ export class PerspectiveElement extends StateElement {
             this.filters = this.getAttribute("filters");
         }
 
+        // Get an edit port from the table, and set it on the viewer so that
+        // all grid edits use the specified port.
+        this._edit_port = await table.make_port();
+
+        // Resolve the edit port lock, which allows for `get_edit_port` to be
+        // called in arbitary order without ever returning a null value.
+        this._edit_port_lock.resolve(this._edit_port);
```

Comment:
I think this is unnecessary if we remove the `await` on line 272 above?  I would also like to understand how this can lead to `null` for port, since it seems (on inspection) that the `Promise` does not defer the actual call to `make_port()`.

I tried removing this on my branch and did not see errors in the tests, but the message implies this error condition is race-y ..